### PR TITLE
Fix the local airflow script

### PR DESCRIPTION
**Pull request by will-cromar**
# Description

Airflow changed the behavior of "allowed_deserialization_classes" in 2.8.1. See https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#allowed-deserialization-classes and https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#allowed-deserialization-classes-regexp

We need to use the "regexp" version after that point, otherwise our dataclasses (e.g. ssh keys) won't deserialize. Update local airflow script accordingly and mark a TODO to make the corresponding change in Terraform when we update Composer.

Also add some additional setup instructions to `contributing`.

# Tests

Tested locally.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run one-shot tests and provided workload links above if applicable. 
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -50,13 +52,15 @@ for this purpose.
 
 ### Testing Changes Locally
 
-To run a dag file in a temporary local environment, use the `local-airflow.sh`:
+To run a dag file in a temporary local environment, use `local-airflow.sh`. The script will symlink just the DAG provided to speed up parsing times.
```
Comment (vanbasten23) [inline docs/contributing.md:55]: maybe also mention it'd better to develop on a cloudtop machine?

Comment (will-cromar) [reply docs/contributing.md:59]: I'll add some instructions to set up VSCode with SSH to cloudtop in our internal onboarding doc.

File: docs/contributing.md [hunk 2]
```diff
@@ -33,6 +33,8 @@ To create a local virtual environment for development, use the [`Python: Create
 
 Once you've set up your virtual environment, ensure you are using the correct intepreter from your local `.venv` directory from the [`Python: Select Interpreter`](https://code.visualstudio.com/docs/python/environments#_select-and-activate-an-environment) command.
 
+When we update the development requirements (especially Airflow itself), you may want to just delete your `.venv` directory and run through these instructions again.
```
Comment (RissyRan) [inline docs/contributing.md:36]: `just delete your `.venv` directory`, will this affect users' other virtual env?

**Review by RissyRan — APPROVED** [review]
Thanks Will!

File: docs/contributing.md (hunk 2 above)
Comment (will-cromar) [reply docs/contributing.md:36]: No, the `.venv` directory is local to this repository, assuming the user set it up with VSCode according to the instructions.

**Pull request merged**